src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1047,13 +1044,15 @@ def _validate_vmss_create_load_balancer_or_app_gateway(cmd, namespace):
             namespace.load_balancer_type = 'new'
             logger.debug('new load balancer will be created')
 
-        if namespace.load_balancer_type == 'new' and not namespace.single_placement_group and std_lb_is_available:
+        if namespace.load_balancer_type == 'new' and namespace.single_placement_group is False and std_lb_is_available:
             LBSkuName = cmd.get_models('LoadBalancerSkuName', resource_type=ResourceType.MGMT_NETWORK)
             if namespace.load_balancer_sku is None:
                 namespace.load_balancer_sku = LBSkuName.standard.value
-                logger.debug("defaulting to Standard Load Balancer for the big scale-set")
+                logger.debug("use Standard sku as single placement group is turned off")
             elif namespace.load_balancer_sku == LBSkuName.basic.value:
-                raise CLIError("usage error: Standard load balancer is required based on other arguments provided")
+                raise CLIError("usage error: 'Standard' load balancer is required because 'single placement group' is "
+                               "turned off to accomodate zonal scale-sets, or with 100+ instances, or Fault Domain "
+                               "count is set to 1")
```

Comment:
It would be better from a user perspective to spell out which of these is the culprit, but this is better than the previous error for sure.